tapfiliate/tapfiliate.php

```diff
@@ -55,84 +58,108 @@ public function uninstall()
+            $state_obj = new State($address_obj->id_state);
+
+            $address1 = $address_obj->address1;
+            $address2 = $address_obj->address2;
+            $city = $address_obj->city;
+            $postcode = $address_obj->postcode;
+            $state = $state_obj->name;
+            $country = $country_obj->iso_code;
+            $vat_number = $address_obj->vat_number;
+            $company = $address_obj->company;
+
+            $api_was_enabled = filter_var(Configuration::get('PS_WEBSERVICE'), FILTER_VALIDATE_BOOLEAN);
+
+            if (!$api_was_enabled) {
+                Configuration::updateValue('PS_WEBSERVICE', 1);
+            }
+
+            $api_key = null;
+            if ($key_id = Configuration::get('TAP_WEBSERVICE_KEY_ID')) {
+                $apiAccess = new WebserviceKey($key_id);
+                $api_key = $apiAccess->key;
+            } else {
+                $apiAccess = new WebserviceKey();
+                $api_key = substr(hash('sha256', uniqid('', true)), 0, 32);
+                $apiAccess->key = $api_key;
+                $apiAccess->save();
+
+                $permissions = [
+                    'customers' => ['GET' => 1, 'POST' => 1, 'PUT' => 1, 'HEAD' => 1],
+                    'orders' => ['GET' => 1, 'POST' => 1, 'PUT' => 1, 'HEAD' => 1],
+                    'configurations' => ['GET' => 1, 'POST' => 1, 'PUT' => 1, 'HEAD' => 1],
+                    'products' => ['GET' => 1, 'POST' => 1, 'PUT' => 1, 'HEAD' => 1],
+                ];
+
+                WebserviceKey::setPermissionForAccount($apiAccess->id, $permissions);
+
+                Configuration::set('TAP_WEBSERVICE_KEY_ID', $apiAccess->id);
+            }
+
+            $payload = [
+                'address1' => $address1,
+                'address2' => $address2,
+                'city' => $city,
+                'postcode' => $postcode,
+                'state' => $state,
+                'country' => $country,
+                'vat_number' => $vat_number,
+                'company' => $company,
+                'domain' => $domain,
+                'email' => $email,
+                'firstname' => $firstname,
+                'lastname' => $lastname,
+                'currency' => $currency->iso_code,
+                'api_key' => $api_key,
+                'shop_id' => $shop_id,
+                'shop_group_id' => $shop_group_id,
+                'login_key' => Configuration::get('TAPFILIATE_LOGIN_KEY')
             ];
 
-            WebserviceKey::setPermissionForAccount($apiAccess->id, $permissions);
-
-            Configuration::set('TAP_WEBSERVICE_KEY_ID', $apiAccess->id);
+            try {
+                $client = new GuzzleHttp\Client();
+                // @TODO remove
+                $client->setDefaultOption('verify', false);
+                $response = $client
+                    ->post(
+                        self::TAPFILIATE_BASE_URL . '/integrations/prestashop/auth/check/',
```

Comment:
Configuration::get('TAPFILIATE_BASE_URL') ?? self::TAPFILIATE_BASE_URL

Reply:
can you clarify please @4overking  `Configuration::get('TAPFILIATE_BASE_URL')`  wherefrom we would take this value? what is the reason to save such value in DB, not in plugin code?